quickwit-cli/src/lib.rs

```diff
@@ -278,14 +278,20 @@ pub async fn index_data_cli(args: IndexDataArgs) -> anyhow::Result<()> {
 
     let source_config_path_opt = args.source_config_path.as_ref();
     let input_path_opt = args.input_path.as_ref();
-    let source_config =
-        create_source_config_from_args(source_config_path_opt, input_path_opt).await?;
+    let mut source_config_res =
+        create_source_config_from_args(source_config_path_opt, input_path_opt).await;
     let indexing_directory_path = args.data_dir_path.join(args.index_id.as_str());
     let indexing_directory = IndexingDirectory::create_in_dir(indexing_directory_path).await?;
     let storage_uri_resolver = quickwit_storage_uri_resolver();
     let metastore_uri_resolver = MetastoreUriResolver::default();
     let metastore = metastore_uri_resolver.resolve(&args.metastore_uri).await?;
+    do_checks([
+        ("config", checked(&mut source_config_res)),
+        ("metastore", metastore.check()),
+        ("index", metastore.check_index(&args.index_id)),
```

Comment:
Cool so you spotted how there was a specific need to check further with `check_index`

Reply:
Yeah, the original issue had an index check, but since the index is now part of the metastore, I added this method.